src/App.tsx

```diff
@@ -1,61 +1,117 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+const todosWithUsers: Todo[] = todosFromServer.map((todo) => {
+  const user = usersFromServer.find((person) => todo.userId === person.id);
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>(todosWithUsers);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [todoUser, setTodoUser] = useState(0);
+  const [todoTitleError, setTodoTitleError] = useState(false);
+  const [todoUserError, setTodoUserError] = useState(false);
+
+  const resetForm = () => {
+    setTodoTitle('');
+    setTodoUser(0);
+  };
+
+  const handleUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setTodoUser(+event.target.value);
+    setTodoUserError(false);
+  };
+
+  const handleTodoTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTodoTitle(event.target.value);
+    setTodoTitleError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim()) {
+      setTodoTitleError(() => true);
+    }
+
+    if (!todoUser) {
+      setTodoUserError(() => true);
+    }
+
+    if (!todoTitle || !todoUser) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: Math.max(...todos.map((todo) => todo.id)) + 1,
+      title: todoTitle,
+      completed: false,
+      userId: todoUser,
+      user: usersFromServer.find((person) => todoUser === person.id),
+    };
+
+    setTodo((prevTodos) => [...prevTodos, newTodo]);
+
+    resetForm();
+  };
+
+  // console.log(todos);
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
```

Comment:
you already have 3 attributes, move them to 3 different lines so its easier to read